src/Stripe.net/Infrastructure/Public/StripeConfiguration.cs

```diff
@@ -49,7 +49,15 @@ public static string ApiKey
                 return apiKey;
             }
 
-            set => apiKey = value;
+            set
+            {
+                if (value != apiKey)
+                {
+                    StripeClient = null;
```

Comment:
Could that affect ongoing requests that rely on this variable? Do we locally cache those or would that suddenly destroy the previously set client?

Reply:
It should be fine:
1. Set `StripeConfiguration.ApiKey = "sk_foo"` --> `StripeConfiguration.StripeClient` is reset to `null`
2. Create a service and send a request. The service will grab the client from `StripeConfiguration.StripeClient`, and the getter will automatically build a new client with `StripeConfiguration.ApiKey`
3. Set `StripeConfiguration.ApiKey = "sk_bar"` --> `StripeConfiguration.StripeClient` is reset to `null`
4. Send another request. The service will grab the client from `StripeConfiguration.StripeClient`, and the getter will automatically build a new client with `StripeConfiguration.ApiKey`

Even in a multi-threaded environment, it should still be fine: the client is not immediately destroyed when `StripeConfiguration.StripeClient` is reset to `null`. If any service has an ongoing request, it will have a reference to the `StripeClient` instance, and the client won't be GC'd until the service is done with the request.

(Really though the proper answer should be "set the global API key _once_ before sending any requests and don't change it afterwards", and we might want to enforce that in the future.)